crates/holochain_terminal/src/client.rs

```diff
@@ -1,25 +1,16 @@
-use anyhow::{anyhow, Context};
+use anyhow::anyhow;
 use holo_hash::{AgentPubKey, DnaHash};
-use holochain_conductor_api::{
-    AdminRequest, AdminResponse, AppAuthenticationRequest, AppAuthenticationToken, AppInfo,
-    AppInterfaceInfo, AppRequest, AppResponse, CellInfo,
-};
+use holochain_client::{AdminWebsocket, AppWebsocket, ClientAgentSigner, DynAgentSigner};
+use holochain_conductor_api::{AppAuthenticationToken, AppInterfaceInfo, CellInfo};
 use holochain_types::network::Kitsune2NetworkMetrics;
 use holochain_types::prelude::InstalledAppId;
 use holochain_types::websocket::AllowedOrigins;
-use holochain_websocket::{connect, ConnectRequest, WebsocketConfig, WebsocketSender};
 use std::collections::HashMap;
-use std::sync::Arc;
 
-pub struct AppClient {
-    tx: WebsocketSender,
-    rx: tokio::task::JoinHandle<()>,
-}
+const HC_TERM_ORIGIN: &str = "hcterm";
 
-impl Drop for AppClient {
-    fn drop(&mut self) {
-        self.rx.abort();
-    }
+pub struct AppClient {
+    client: AppWebsocket,
```

Comment:
I found these `AppClient` and `AdminClient` wrappers still quite handy actually but I'm also fine to be told to remove them and move the code that their methods contain into separate functions or inline where required.

Reply:
Nothing wrong with that, they're exposing an interface that is useful for what this app is doing. If it's still useful then we should keep them :)